test/Spaces/terrain_warp.jl

```diff
@@ -124,17 +161,61 @@ end
         polynom = 2:2:10
         horzelem = 2:2:10
         for nl in levels, np in polynom, nh in horzelem
+            ʷhv_center_space, ʷhv_face_space =
+                warpedspace_2D(FT, (xmin, xmax), (zmin, zmax), nh, nl, np;)
             hv_center_space, hv_face_space =
-                hvspace_2D(FT, (xmin, xmax), (zmin, zmax), nh, nl, np;)
-            ᶜcoords = Fields.coordinate_field(hv_center_space)
-            ᶠcoords = Fields.coordinate_field(hv_face_space)
-            z₀ = ClimaCore.Fields.level(ᶜcoords.z, 1)
+                hybridspace_2D(FT, (xmin, xmax), (zmin, zmax), nh, nl, np)
+            ⁿhv_center_space, ⁿhv_face_space = warpedspace_2D(
+                FT,
+                (xmin, xmax),
+                (zmin, zmax),
+                nh,
+                nl,
+                np;
+                warp_fn = warp_test_2d,
+            )
+            ʷᶜcoords = Fields.coordinate_field(ʷhv_center_space)
+            ʷᶠcoords = Fields.coordinate_field(ʷhv_face_space)
+
+            z₀ = ClimaCore.Fields.level(ʷᶜcoords.z, 1)
             # Check ∫ₓ(z_sfc)dx == known value from warp_test_2d
             @test sum(z₀ .- zmax / 2nl) - FT(1) <= FT(0.1 / np * nh * nl)
             @test abs(maximum(z₀) - FT(0.5)) <= FT(0.125)
         end
     end
 end
+
+@testset "2D Warped Mesh RHS Integration Test" begin
+    for FT in (Float64,)
+        xmin, xmax = FT(0), FT(π)
+        zmin, zmax = FT(0), FT(1)
+        levels = 10
+        polynom = 4
+        horzelem = 5
+        ⁿhv_center_space, ⁿhv_face_space = warpedspace_2D(
+            FT,
+            (xmin, xmax),
+            (zmin, zmax),
+            horzelem,
+            levels,
+            polynom;
+            warp_fn = nowarp_test_2d,
+        )
+        ⁿᶜcoords = Fields.coordinate_field(ⁿhv_center_space)
+        ⁿᶠcoords = Fields.coordinate_field(ⁿhv_face_space)
+        uₕ = map(_ -> Geometry.Covariant1Vector(1.0), ⁿᶜcoords)
+        w = map(_ -> Geometry.Covariant3Vector(0.0), ⁿᶠcoords)
+        Y = Fields.FieldVector(uₕ = uₕ, w = w)
+        dY = similar(Y)
+        function rhs!(dY, Y, _, t)
+            Spaces.weighted_dss!(dY.uₕ)
+            Spaces.weighted_dss!(dY.w)
```

Comment:
This is an attempt to set up a unit test with the minimum working components from the driver / simulation file `rhs` evaluations.